executor/test/showtest/show_test.go

```diff
@@ -2082,3 +2082,35 @@ func TestShowBindingDigestField(t *testing.T) {
 	result = tk.MustQuery("show global bindings;")
 	require.Equal(t, len(result.Rows()), 0)
 }
+
+func TestShowPasswordVariable(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+
+	tk.MustExec("SET GLOBAL authentication_ldap_sasl_bind_root_pwd = ''")
+	rs, err := tk.Exec("show variables like 'authentication_ldap_sasl_bind_root_pwd'")
+	require.NoError(t, err)
+	require.Equal(t, tk.ResultSetToResult(rs, "").Rows()[0][1], "")
+
+	tk.MustExec("SET GLOBAL authentication_ldap_sasl_bind_root_pwd = password")
+	defer func() {
+		tk.MustExec("SET GLOBAL authentication_ldap_sasl_bind_root_pwd = ''")
+	}()
+	rs, err = tk.Exec("show variables like 'authentication_ldap_sasl_bind_root_pwd'")
+	require.NoError(t, err)
+	require.Equal(t, tk.ResultSetToResult(rs, "").Rows()[0][1], "******")
+
+	tk.MustExec("SET GLOBAL authentication_ldap_simple_bind_root_pwd = ''")
+	rs, err = tk.Exec("show variables like 'authentication_ldap_simple_bind_root_pwd'")
+	require.NoError(t, err)
+	require.Equal(t, tk.ResultSetToResult(rs, "").Rows()[0][1], "")
+
+	tk.MustExec("SET GLOBAL authentication_ldap_simple_bind_root_pwd = password")
+	defer func() {
+		tk.MustExec("SET GLOBAL authentication_ldap_simple_bind_root_pwd = ''")
+	}()
+
+	rs, err = tk.Exec("show variables like 'authentication_ldap_simple_bind_root_pwd'")
+	require.NoError(t, err)
+	require.Equal(t, tk.ResultSetToResult(rs, "").Rows()[0][1], "******")
```

Comment:
Please add a test about `information_schema.VARIABLES_INFO`, the current value of it should also be masked. I think this PR is good enough to handle this case, just adding a new test is enough to avoid breaking the behavior in the future.

There is another table `mysql.GLOBAL_VARIABLES` which isn't masked. This PR could ignore this problem (if it is) as we have a lot of workarounds, and it's pretty hard to handle this right.